pkgs/blast_repo/lib/src/tweaks/drop_lint_tweak.dart

```diff
@@ -0,0 +1,88 @@
+// Copyright (c) 2022, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:io';
+
+import 'package:collection/collection.dart';
+import 'package:path/path.dart' as p;
+import 'package:yaml/yaml.dart';
+import 'package:yaml_edit/yaml_edit.dart';
+
+import '../repo_tweak.dart';
+
+final _instance = DropLintTweak._();
+
+class DropLintTweak extends RepoTweak {
+  factory DropLintTweak() => _instance;
+
+  DropLintTweak._()
+      : super(
+          id: 'drop-lint',
+          description: 'Drop deprecated lints from analysis_options.yaml',
+        );
+
+  @override
+  bool shouldRunByDefault(Directory checkout, String repoSlug) => true;
+
+  @override
+  FutureOr<FixResult> fix(Directory checkout, String repoSlug) async {
+    final analysisOptionsFile =
+        File(p.join(checkout.path, 'analysis_options.yaml'));
+
+    if (!analysisOptionsFile.existsSync()) {
+      return FixResult.noFixesMade;
+    }
+
+    final yamlSource = loadYaml(
+      analysisOptionsFile.readAsStringSync(),
+      sourceUrl: analysisOptionsFile.uri,
+    );
+
+    if (yamlSource is YamlMap) {
+      final linterNode = yamlSource['linter'];
+      if (linterNode is YamlMap) {
+        final rules = linterNode['rules'];
+
+        if (rules is YamlList) {
```

Comment:
There is almost certainly some magic case expression thingy I could do here...but I'm not smart enough.

CC @munificent